genericfilestorage/storage.go

```diff
@@ -0,0 +1,105 @@
+package genericfilestorage
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+)
+
+// FileProvider ...
+type FileProvider interface {
+	GetFiles() ([]File, error)
+}
+
+// BulkFileDownloader ...
+type BulkFileDownloader interface {
+	DownloadFiles(files []File, targetDir string) error
+}
+
+// Storage ...
+type Storage struct {
+	bulkdownloader BulkFileDownloader
+	fileprovider   FileProvider
+	envKey         string
+}
+
+// NewStorage ...
+func NewStorage(bulkdownloader BulkFileDownloader,
+	fileprovider FileProvider,
+	envKey string) Storage {
+	return Storage{
+		bulkdownloader: bulkdownloader,
+		fileprovider:   fileprovider,
+		envKey:         envKey,
+	}
+}
+
+// DownloadFiles ...
+func (storage Storage) DownloadFiles() (string, error) {
+
+	localStorageDir, err := storage.createLocalStorageDir()
+	if err != nil {
+		return "", err
+	}
+
+	files, err := storage.filesToDownload()
+	if err != nil {
+		return "", err
+	}
+
+	fmt.Println()
+	log.Infof("Downloading %d files:", len(files))
```

Comment:
We could but it's still clear from the logs that there were 0 files to download. Plus there is not much performance cost of executing the next few lines. I would keep it as is, as this would rarely be the case that the step is added but there is no file to download.

Reply:
Yeah, it's just that it looks a bit weird to say `"Downloading 0 files:"`. But I'm ok with leaving it here